blog-content/aws-firehose-opentelemetry-collector.md

```diff
@@ -0,0 +1,183 @@
+
+In this guide, we’ll explore how to configure the OpenTelemetry Collector to receive and process data from AWS Firehose.
+
+## AWS Firehose
+
+Before configuring OpenTelemetry, it’s important to understand the format of data sources that Firehose supports. AWS Firehose typically receives data from services like CloudWatch Metrics and CloudWatch Logs.
+
+## Firehose Sources
+
+### CloudWatch Metrics Stream Format
+
+AWS Firehose can be configured to receive CloudWatch metrics in either OpenTelemetry 1.0 format or JSON format. The key difference is:
+•	OpenTelemetry 1.0 format: Structured for direct ingestion into observability platforms.
+•	Metrics JSON format: A more generic format that requires custom parsing before ingestion into OpenTelemetry.
+
+Example JSON payload from a CloudWatch Metrics stream:
+
+```json
+{
+    "namespace": "AWS/EC2",
+    "metric_name": "CPUUtilization",
+    "dimensions": {
+    "InstanceId": "i-1234567890abcdef0"
+    },
+    "timestamp": 1706000000,
+    "value": 23.5
+}
+```
+
+To correctly process this data in OpenTelemetry, a receiver must be configured to parse and extract relevant metrics.
+
+### CloudWatch Logs Format
+
+CloudWatch logs can also be streamed to Firehose, where they are typically formatted in JSON. A sample log entry might look like this:
+
+```json
+{
+    "owner": "123456789012",
+    "logGroup": "/aws/lambda/my-function",
+    "logStream": "2025/01/23/[$LATEST]abcdef1234567890",
+    "message": "Function executed successfully",
+    "timestamp": 1706000000
+}
+```
+
+These logs need to be parsed properly to extract meaningful telemetry data before ingestion into OpenTelemetry.
+
+## OpenTelemetry Collector
+
+The OpenTelemetry Collector is a vendor-agnostic proxy for receiving, processing, and exporting telemetry data. To integrate Firehose, you need to use a custom receiver that can interpret Firehose-formatted payloads.
+
+## Custom Receivers
+
+The OpenTelemetry Collector supports custom receivers through the opentelemetry-collector-contrib repository. The Firehose receiver is not part of the core OpenTelemetry distribution, so you must use the contrib image.
+
+To use the contrib image, modify your OpenTelemetry Collector configuration as follows:
+
+```yaml
+receivers:
+    awsfirehose/cwmetrics:
+        endpoint: '0.0.0.0:4433'
+        record_type: cwmetrics
+        include_metadata: true
+    awsfirehose/cwlogs:
+        endpoint: '0.0.0.0:4434'
+        record_type: cwlogs
+        include_metadata: true
+    awsfirehose/otlp_v1:
+        endpoint: '0.0.0.0:4435'
+        record_type: otlp_v1
+        include_metadata: true
+    
+exporters:
+    logging:
+    verbosity: detailed
+    
+service:
+    pipelines:
+        logs:
+        receivers: [awsfirehose/cwmetrics, awsfirehose/cwlogs, awsfirehose/otlp_v1]
+        exporters: [logging]
+```
+
+This configuration sets up a Firehose receiver that listens on port 8000 and processes logs in JSON format.
```

Comment:
Is this right? Aren't we using different ports to support all different metric types? Do we want to explain what each receiver is doing?

Reply:
Also, is the `logging` exporter just for debugging? What do you think about using an otlp exporter to make it more like a real-world config?

Reply:
yeah good idea

Reply:
added!